lib/driver.js

```diff
@@ -1,407 +1,388 @@
+        factor: 1,
+        minTimeout: retryDelay,
+        maxTimeout: retryDelay,
+        randomize: false
+      };
+    }
+    config.supportsPreparedStatements = true;
+    this.config = config;
+  }
 
-  this.pools = {};
-  this.resultTransformers = context.resultTransformers || [];
-};
+  _initProviderOptions() {
+    setHelenusOptions(this.config);
+  }
 
-DatastaxDriver.prototype.initProviderOptions = function init(config) {
-  setHelenusOptions(config);
-  config.supportsPreparedStatements = true;
-};
+  _initPoolConfig() {
+    this.poolConfig = {
+      consistencyLevel: this.consistencyLevel.one
+    };
+    this.poolConfig = _.extend(this.poolConfig, this.config);
 
-DatastaxDriver.prototype.initPoolConfig = function () {
-  this.poolConfig = {
-    consistencyLevel: this.consistencyLevel.one
-  };
-  this.poolConfig = _.extend(this.poolConfig, this.config);
+    if (typeof this.config.consistency === 'string') {
+      var level = this.config.consistency;
 
-  if (typeof this.config.consistency === 'string') {
-    var level = this.config.consistency;
+      if (typeof this.consistencyLevel[level] !== 'undefined') {
+        this.poolConfig.consistencyLevel = this.consistencyLevel[level];
+      } else {
+        throw new Error('Error: "' + level + '" is not a valid consistency level');
+      }
+    }
 
-    if (typeof this.consistencyLevel[level] !== 'undefined') {
-      this.poolConfig.consistencyLevel = this.consistencyLevel[level];
-    } else {
-      throw new Error('Error: "' + level + '" is not a valid consistency level');
+    if (typeof this.config.localDataCenter === 'string') {
+      this.poolConfig.policies = this.poolConfig.policies || {};
+      if (!this.poolConfig.policies.loadBalancing) {
+        this.poolConfig.policies.loadBalancing = new cqlDriver.policies.loadBalancing.TokenAwarePolicy(
+          new cqlDriver.policies.loadBalancing.DCAwareRoundRobinPolicy(this.config.localDataCenter)
+        );
+      }
     }
   }
 
-  if (typeof this.config.localDataCenter === 'string') {
-    this.poolConfig.policies = this.poolConfig.policies || {};
-    if (!this.poolConfig.policies.loadBalancing) {
-      this.poolConfig.policies.loadBalancing = new cqlDriver.policies.loadBalancing.TokenAwarePolicy(
-        new cqlDriver.policies.loadBalancing.DCAwareRoundRobinPolicy(this.config.localDataCenter)
-      );
+  _initConnectionResolver(context) {
+    this.connectionResolver = null;
+    if (context.connectionResolver) {
+      this.connectionResolver = context.connectionResolver;
+    } else if (this.config.connectionResolverPath) {
+      this.connectionResolver = require(this.config.connectionResolverPath);
+    }
+    if (this.connectionResolver) {
+      if (typeof this.connectionResolver === 'function') {
+        this.connectionResolver = this.connectionResolver();
+      }
+      /* istanbul ignore else: no benefit of testing event handlers not being wired */
```

Comment:
I don't think we care about 100% coverage, so I'd be fine removing these sort of statements.

Reply:
@scommisso: removed, and also brought coverage up to 100% for the heck of it through more tests and removing confirmed dead code.